language/tools/move-cli/src/sandbox/commands/publish.rs

```diff
@@ -115,8 +116,16 @@ pub fn publish(
                     let res =
                         session.publish_module_bundle(module_bytes_vec, sender, &mut gas_status);
                     if let Err(err) = res {
-                        // TODO (mengxu): explain publish errors in multi-module publishing
                         println!("Invalid multi-module publishing: {}", err);
+                        if let Location::Module(module_id) = err.location() {
+                            // find the module where error occures and explain
+                            if let Some(unit) = modules_to_publish
+                                .into_iter()
+                                .find(|&x| x.unit.name().as_str() == module_id.name().as_str())
+                            {
+                                explain_publish_error(err, state, unit)?
+                            }
```

Comment:
I would probably leave a println on the else path as well, with a message like "Unable to locate the module in the multi-module publishing error"